Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -271,7 +271,11 @@ private async Task<Exception> InvokeWithInterceptionAsync(TimeSpan timeout)
                 // Here we do not need to know whether the task completes (successfully) in timeout
                 // or does not complete. We are only interested in the exception which is thrown, not returned.
                 // So, we can ignore the result.
-                _ = await CompletesWithinTimeoutAsync(task, remainingTime);
+                _ = await CompletesWithinTimeoutAsync(task, remainingTime, async cancelledTask =>
+                {
+                    // Rethrow the exception causing the task be canceled.
+                    await cancelledTask;
+                });
```

Comment:
If we remove the `async`+`await` the stack trace will look a bit nicer (in my opinion).

with async+await
```
at FluentAssertions.Specialized.AsyncFunctionAssertions`2.<>c.<<InvokeWithInterceptionAsync>b__10_0>d.MoveNext() in C:\dev\fluentassertions\Src\FluentAssertions\Specialized\AsyncFunctionAssertions.cs:line 276
at FluentAssertions.Specialized.AsyncFunctionAssertions`2.<CompletesWithinTimeoutAsync>d__15.MoveNext() in C:\dev\fluentassertions\Src\FluentAssertions\Specialized\AsyncFunctionAssertions.cs:line 458
```

without async+await
```
at FluentAssertions.Specialized.AsyncFunctionAssertions`2.<CompletesWithinTimeoutAsync>d__15.MoveNext() in C:\dev\fluentassertions\Src\FluentAssertions\Specialized\AsyncFunctionAssertions.cs:line 458
```




```suggestion
                _ = await CompletesWithinTimeoutAsync(task, remainingTime, cancelledTask =>
                {
                    // Rethrow the exception causing the task be canceled.
                    return cancelledTask;
                });
```